parsers/ini/ini.go

```diff
@@ -0,0 +1,109 @@
+package ini
+
+import (
+	"fmt"
+	"gopkg.in/ini.v1"
+	"strings"
+	"time"
+)
+
+type INI struct {}
+
+func Parser() *INI {
+	return &INI{}
+}
+
+func (p *INI) Unmarshal(b []byte) (map[string]interface{}, error) {
+	out := make(map[string]interface{}, 0)
+
+	iniFile, err := ini.Load(b)
+	if err != nil {
+		fmt.Println("ini.Load error")
+		return nil, err
+	}
+
+	for _, section := range iniFile.Sections() {
+		if strings.Compare(section.Name(), "DEFAULT") == 0 {
+			for _, key := range section.Keys() {
+				out[key.Name()] = convertKey(key)
+			}
+		} else {
+			mpS := make(map[string]interface{}, 0)
+			for _, key := range section.Keys() {
+				mpS[key.Name()] = convertKey(key)
+			}
+			out[section.Name()] = mpS
+		}
+	}
+
+	return out, nil
+}
+
+func (p *INI) Marshal(o map[string]interface{}) ([]byte, error) {
```

Comment:
`go-ini` documentation says that it supports reading/parsing and writing also. This custom marshaling logic isn't required.

Should be as simple as using `WriteTo()` to write INI bytes into a byte buffer and returning that.

https://pkg.go.dev/github.com/go-ini/ini#File.WriteTo

Reply:
go-ini cannot work with maps. It was the main problem, when I tried to adapt it to koanf.